crates/ruff_linter/src/rules/ruff/rules/split_of_static_string.rs

```diff
@@ -0,0 +1,166 @@
+                    value: StringLiteralValue::single(StringLiteral {
+                        value: (*list_item).to_string().into_boxed_str(),
+                        range: TextRange::default(),
+                        flags: StringLiteralFlags::default(),
+                    }),
+                    range: TextRange::default(),
+                })
+            })
+            .collect(),
+        ctx: ExprContext::Load,
+        range: TextRange::default(),
+    })
+}
+
+fn split_default(str_value: &str, max_split: usize) -> Option<Expr> {
+    // From the Python documentation:
+    // > If sep is not specified or is None, a different splitting algorithm is applied: runs of
+    // > consecutive whitespace are regarded as a single separator, and the result will contain
+    // > no empty strings at the start or end if the string has leading or trailing whitespace.
+    // > Consequently, splitting an empty string or a string consisting of just whitespace with
+    // > a None separator returns [].
+    // https://docs.python.org/3/library/stdtypes.html#str.split
+    if max_split == 0 {
+        let list_items: Vec<&str> = str_value.split_whitespace().collect();
+        Some(construct_replacement(&list_items))
+    } else {
+        // Autofix for maxsplit without separator not yet implemented
+        None
+    }
+}
+
+fn split_sep(str_value: &str, sep_value: &str, max_split: usize, direction_left: bool) -> Expr {
+    let list_items: Vec<&str> = if direction_left && max_split > 0 {
+        str_value.splitn(max_split + 1, sep_value).collect()
+    } else if !direction_left && max_split > 0 {
+        str_value.rsplitn(max_split + 1, sep_value).collect()
+    } else if direction_left && max_split == 0 {
+        str_value.split(sep_value).collect()
+    } else {
+        str_value.rsplit(sep_value).collect()
+    };
+    construct_replacement(&list_items)
+}
+
+/// RUF035
+pub(crate) fn split_of_static_string(
+    checker: &mut Checker,
+    attr: &str,
+    call: &ExprCall,
+    str_value: &str,
+) {
+    let ExprCall { arguments, .. } = call;
+
+    let sep_arg = arguments.find_argument("sep", 0);
+    let maxsplit_arg = arguments.find_argument("maxsplit", 1);
+
+    // `split` vs `rsplit`
+    let direction_left = attr == "split";
+
+    let maxsplit_value = if let Some(maxsplit) = maxsplit_arg {
+        match maxsplit {
+            Expr::NumberLiteral(maxsplit_val) => {
+                if let Some(int_value) = maxsplit_val.value.as_int() {
+                    if let Some(usize_value) = int_value.as_usize() {
+                        usize_value
+                    } else {
+                        return;
+                    }
+                } else {
+                    return;
+                }
+            }
+            // Ignore when `maxsplit` is not a numeric value
+            _ => {
+                return;
+            }
+        }
+    } else {
+        0
+    };
```

Comment:
Nit: You could use `Option::and_then` to reduce the nesting

```suggestion
    let maxsplit_value = if let Some(maxsplit) = maxsplit_arg {
        let Some(value) = maxsplit
            .as_number_literal_expr()
            .and_then(|max_split| max_split.value.as_int())
            .and_then(|int_value| int_value.as_usize())
        else {
            return;
        };

        value
    } else {
        0
    };
```

Reply:
I kept the match statement now that UnaryOp is also matched, but removed one if-else condition with `and_then`.